contracts/src/v0.8/automation/dev/test/AutomationRegistry2_2.t.sol

```diff
@@ -0,0 +1,189 @@
+
+contract AutomationRegistry2_2_SetUp is BaseTest {
+  address internal constant LINK_ETH_FEED = 0x1111111111111111111111111111111111111110;
+  address internal constant FAST_GAS_FEED = 0x1111111111111111111111111111111111111112;
+  address internal constant LINK_TOKEN = 0x1111111111111111111111111111111111111113;
+
+  // Signer private keys used for these test
+  uint256 internal constant PRIVATE0 = 0x7b2e97fe057e6de99d6872a2ef2abf52c9b4469bc848c2465ac3fcd8d336e81d;
+  uint256 internal constant PRIVATE1 = 0xab56160806b05ef1796789248e1d7f34a6465c5280899159d645218cd216cee6;
+  uint256 internal constant PRIVATE2 = 0x6ec7caa8406a49b76736602810e0a2871959fbbb675e23a8590839e4717f1f7f;
+  uint256 internal constant PRIVATE3 = 0x80f14b11da94ae7f29d9a7713ea13dc838e31960a5c0f2baf45ed458947b730a;
+
+  uint64 internal constant OFFCHAIN_CONFIG_VERSION = 30; // 2 for OCR2
+  uint8 internal constant F = 1;
+
+  address[] internal s_valid_signers;
+  address[] internal s_valid_transmitters;
+  address[] internal s_registrars;
+
+  function setUp() public override {
+    s_valid_transmitters = new address[](4);
+    for (uint160 i = 0; i < 4; ++i) {
+      s_valid_transmitters[i] = address(4 + i);
+    }
+
+    s_valid_signers = new address[](4);
+    s_valid_signers[0] = vm.addr(PRIVATE0); //0xc110458BE52CaA6bB68E66969C3218A4D9Db0211
+    s_valid_signers[1] = vm.addr(PRIVATE1); //0xc110a19c08f1da7F5FfB281dc93630923F8E3719
+    s_valid_signers[2] = vm.addr(PRIVATE2); //0xc110fdF6e8fD679C7Cc11602d1cd829211A18e9b
+    s_valid_signers[3] = vm.addr(PRIVATE3); //0xc11028017c9b445B6bF8aE7da951B5cC28B326C0
+
+    s_registrars = new address[](1);
+    s_registrars[0] = 0x3a0eDE26aa188BFE00b9A0C9A431A1a0CA5f7966;
+  }
+
+  function deployRegistry2_2(AutomationRegistryBase2_2.Mode mode) public returns (IAutomationRegistryMaster) {
+    AutomationForwarderLogic forwarderLogic = new AutomationForwarderLogic();
+    AutomationRegistryLogicB2_2 logicB2_2 = new AutomationRegistryLogicB2_2(
+      mode,
+      LINK_TOKEN,
+      LINK_ETH_FEED,
+      FAST_GAS_FEED,
+      address(forwarderLogic)
+    );
+    AutomationRegistryLogicA2_2 logicA2_2 = new AutomationRegistryLogicA2_2(logicB2_2);
+    IAutomationRegistryMaster registry2_2 = IAutomationRegistryMaster(
+      address(new AutomationRegistry2_2(AutomationRegistryLogicB2_2(address(logicA2_2))))
+    );
+    return registry2_2;
+  }
+}
+
+contract AutomationRegistry2_2_LatestConfigDetails is AutomationRegistry2_2_SetUp {
+  function testGet() public {
+    IAutomationRegistryMaster registry = IAutomationRegistryMaster(
+      address(deployRegistry2_2(AutomationRegistryBase2_2.Mode(0)))
+    );
+    (uint32 configCount, uint32 blockNumber, bytes32 configDigest) = registry.latestConfigDetails();
+    assertEq(configCount, 0);
+    assertEq(blockNumber, 0);
+    assertEq(configDigest, "");
+  }
+}
+
+contract AutomationRegistry2_2_SetConfig is AutomationRegistry2_2_SetUp {
```

Comment:
I think we can keep all of the tests in a single `AutomationRegistry2_2_Test` contract, yeah? I don't see the need to separate tests into their own contracts